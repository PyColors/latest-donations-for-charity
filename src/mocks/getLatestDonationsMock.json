{  
   "name":"Macmillan Cancer Support",
   "description":"At Macmillan, we know cancer can affect everything. Your body, your relationships, your money, everything. But Macmillan is here to help people live life, no matter what. We give you the support to hold on to who you are and what’s important. Life with cancer is still life –we’ll help you live it.",
   "logoUrl":"e419bab6-f588-467f-a64e-5005eeb0734b.jpg",
   "logoAbsoluteUrl":"https://images.justgiving.com/image/e419bab6-f588-467f-a64e-5005eeb0734b.jpg",
   "profilePageUrl":"https://www.justgiving.com/macmillan",
   "registrationNumber":"261017",
   "websiteUrl":"http://www.macmillan.org.uk",
   "id":2116,
   "pageShortName":"macmillan",
   "emailAddress":"fundraising@macmillan.org.uk",
   "keywords":"MCR",
   "dateAddedToJustGiving":"\/Date(1036700780000+0000)\/",
   "thankyouMessage":"Thanks so much for your donation to Macmillan Cancer Support. Your support means an incredible amount to us.\r\n\r\nAt Macmillan, we know cancer can affect everything. Your body, your relationships, your money, everything. But Macmillan is here to help people live life, no matter what. We give them the support they need to hold on to who they are and what’s important.  But we can’t do it without you. \r\n\r\nLife with cancer is still life – let’s help people live it.",
   "mobileAppeals":[  
      {  
         "smsCode":"MOBL11",
         "name":"Mob Web donations 2011"
      },
      {  
         "smsCode":"IMEM01",
         "name":"Mac In Memory Campaign"
      },
      {  
         "smsCode":"SUSX13",
         "name":"Sussex Appeal"
      },
      {  
         "smsCode":"MACS12",
         "name":"Mac Social Media 2012"
      },
      {  
         "smsCode":"LINK25",
         "name":"Translink partnership"
      },
      {  
         "smsCode":"BELF01",
         "name":"Belfast takeover"
      },
      {  
         "smsCode":"MACM14",
         "name":"City of London Takeover"
      },
      {  
         "smsCode":"MANC14",
         "name":"Manchester city FR'ing"
      },
      {  
         "smsCode":"TAME16",
         "name":"Tameside Macmillan Unit"
      }
   ],
   "categories":[  
      "Cancer"
   ],
   "donationDisplayAmounts":[  
      {  
         "amount":3.00,
         "description":"Can help answer questions on cancer and give support through our helpline",
         "name":"£3.00",
         "donationPromptType":"Monthly"
      },
      {  
         "amount":25.00,
         "description":"Can fund an hour on our support line, giving vital information and support",
         "name":"£25.00",
         "donationPromptType":"OneOff"
      },
      {  
         "amount":5.00,
         "description":"Can help provide counselling to someone dealing with the impact of cancer",
         "name":"£5.00",
         "donationPromptType":"Monthly"
      },
      {  
         "amount":50.00,
         "description":"Can ensure a Macmillan Nurse is there when someone is diagnosed with cancer",
         "name":"£50.00",
         "donationPromptType":"OneOff"
      },
      {  
         "amount":10.00,
         "description":"Can help to provide the one to one medical care which matters so much",
         "name":"£10.00",
         "donationPromptType":"Monthly"
      },
      {  
         "amount":100.00,
         "description":"Can provide benefit advice so people with cancer get the best deal possible",
         "name":"£100.00",
         "donationPromptType":"OneOff"
      }
   ],
   "impactStatementWhat":"know cancer affects everything and exist",
   "impactStatementWhy":"help you live your life.",
   "countryCode":"GB",
   "currencyCode":"GBP",
   "themeColour":{  
      "alpha":255,
      "red":0,
      "green":162,
      "blue":70
   },
   "smsShortName":"Macmillan Cancer Support"
}