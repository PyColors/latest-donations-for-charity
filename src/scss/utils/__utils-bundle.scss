/*
This file is used to contain all utils imports.
*/

//Import Utils files

@import 'fonts';
@import 'mixins';
@import 'variables';