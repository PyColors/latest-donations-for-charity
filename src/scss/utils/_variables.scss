
$primary-color: #00a246;
$secondary-color: #fff;
$tertiary-color: #8CC63F;


$white: #fff; 
$grey-light: #EBEBEB;
$grey: #cacaca;

$radius-button: 10px;