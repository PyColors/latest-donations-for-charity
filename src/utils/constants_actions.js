/* ACTIONS */

/* Common */
export const CLEAR_STORE = 'CLEAR_STORE';

/* Home Page */
export const FETCH_DATA_HOME_PAGE = "FETCH_DATA_HOME_PAGE";
export const RECEIVE_DATA_HOME_PAGE = "RECEIVE_DATA_HOME_PAGE";
export const FAILED_RECIEVE_DATA_HOME_PAGE = "FAILED_RECIEVE_DATA_HOME_PAGE";
