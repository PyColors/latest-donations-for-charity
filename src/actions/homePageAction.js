import * as constantsActions from "../utils/constants_actions";
import axios from "axios";

export function fetchDataIfNeeded() {
  return (dispatch, getState) => {
    if (shouldFetchData(getState())) {
      return dispatch(fetchData());
    }
  };
}

export function fetchData() {
  return dispatch => {
    dispatch(beginFetchData());
    axios
      .get(`https://api.justgiving.com/2e610e0e/v1/charity/2116`)
      .then(res => {
        dispatch(loadDataHomePage(res));
      })
      .catch(error => {
        const response = error.response;

        if (response === undefined) {
          dispatch(fetchFailure(error));
        } else {
          error.status = response.status;
          error.statusText = response.statusText;
          response.text().then(text => {
            try {
              const json = JSON.parse(text);
              error.message = json.message;
            } catch (ex) {
              error.message = text;
            }
            dispatch(fetchFailure(error));
          });
        }
      });
  };
}

function shouldFetchData(state) {
  if (state === null) {
    return false;
  } else {
    return true;
  }
}

const beginFetchData = () => ({
  type: constantsActions.FETCH_DATA_HOME_PAGE
});

const loadDataHomePage = payload => ({
  type: constantsActions.RECEIVE_DATA_HOME_PAGE,
  dataHomePage: payload.data
});

const fetchFailure = () => ({
  type: constantsActions.FAILED_RECIEVE_DATA_HOME_PAGE,
  error: true
});
