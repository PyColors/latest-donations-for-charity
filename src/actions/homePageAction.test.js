import configureMockStore from "redux-mock-store";

import thunk from "redux-thunk";
import moxios from "moxios";
import expect from "expect";
import * as constantsActions from "../utils/constants_actions";
import getLatestDonationsMock from "../mocks/getLatestDonationsMock";
import {fetchData, loadDataHomePage} from './homePageAction'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe("fetchData actions", () => {
    beforeEach(function() {
        moxios.install();
    });

    afterEach(function() {
        moxios.uninstall();
    });

  describe("Actions", () => {
        moxios.wait(() => {
            const request = moxios.requests.mostRecent();
            request.respondWith({
                status: 200,
                response: getLatestDonationsMock
            });
        });

    it('Return dataHomePage', () => {
      const expectedData = { dataHomePage: getLatestDonationsMock }
      const store = mockStore({
        dataHomePage: getLatestDonationsMock
      })
      expect(store.getState(fetchData)).toEqual(expectedData);
      expect(store.getState()).toMatchSnapshot();
    });
  });
});

