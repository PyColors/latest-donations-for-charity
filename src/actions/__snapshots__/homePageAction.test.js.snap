// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchData actions Actions Data 1`] = `
Object {
  "dataHomePage": Object {
    "categories": Array [
      "Cancer",
    ],
    "countryCode": "GB",
    "currencyCode": "GBP",
    "dateAddedToJustGiving": "/Date(1036700780000+0000)/",
    "description": "At Macmillan, we know cancer can affect everything. Your body, your relationships, your money, everything. But Macmillan is here to help people live life, no matter what. We give you the support to hold on to who you are and what’s important. Life with cancer is still life –we’ll help you live it.",
    "donationDisplayAmounts": Array [
      Object {
        "amount": 3,
        "description": "Can help answer questions on cancer and give support through our helpline",
        "donationPromptType": "Monthly",
        "name": "£3.00",
      },
      Object {
        "amount": 25,
        "description": "Can fund an hour on our support line, giving vital information and support",
        "donationPromptType": "OneOff",
        "name": "£25.00",
      },
      Object {
        "amount": 5,
        "description": "Can help provide counselling to someone dealing with the impact of cancer",
        "donationPromptType": "Monthly",
        "name": "£5.00",
      },
      Object {
        "amount": 50,
        "description": "Can ensure a Macmillan Nurse is there when someone is diagnosed with cancer",
        "donationPromptType": "OneOff",
        "name": "£50.00",
      },
      Object {
        "amount": 10,
        "description": "Can help to provide the one to one medical care which matters so much",
        "donationPromptType": "Monthly",
        "name": "£10.00",
      },
      Object {
        "amount": 100,
        "description": "Can provide benefit advice so people with cancer get the best deal possible",
        "donationPromptType": "OneOff",
        "name": "£100.00",
      },
    ],
    "emailAddress": "fundraising@macmillan.org.uk",
    "id": 2116,
    "impactStatementWhat": "know cancer affects everything and exist",
    "impactStatementWhy": "help you live your life.",
    "keywords": "MCR",
    "logoAbsoluteUrl": "https://images.justgiving.com/image/e419bab6-f588-467f-a64e-5005eeb0734b.jpg",
    "logoUrl": "e419bab6-f588-467f-a64e-5005eeb0734b.jpg",
    "mobileAppeals": Array [
      Object {
        "name": "Mob Web donations 2011",
        "smsCode": "MOBL11",
      },
      Object {
        "name": "Mac In Memory Campaign",
        "smsCode": "IMEM01",
      },
      Object {
        "name": "Sussex Appeal",
        "smsCode": "SUSX13",
      },
      Object {
        "name": "Mac Social Media 2012",
        "smsCode": "MACS12",
      },
      Object {
        "name": "Translink partnership",
        "smsCode": "LINK25",
      },
      Object {
        "name": "Belfast takeover",
        "smsCode": "BELF01",
      },
      Object {
        "name": "City of London Takeover",
        "smsCode": "MACM14",
      },
      Object {
        "name": "Manchester city FR'ing",
        "smsCode": "MANC14",
      },
      Object {
        "name": "Tameside Macmillan Unit",
        "smsCode": "TAME16",
      },
    ],
    "name": "Macmillan Cancer Support",
    "pageShortName": "macmillan",
    "profilePageUrl": "https://www.justgiving.com/macmillan",
    "registrationNumber": "261017",
    "smsShortName": "Macmillan Cancer Support",
    "thankyouMessage": "Thanks so much for your donation to Macmillan Cancer Support. Your support means an incredible amount to us.

At Macmillan, we know cancer can affect everything. Your body, your relationships, your money, everything. But Macmillan is here to help people live life, no matter what. We give them the support they need to hold on to who they are and what’s important.  But we can’t do it without you. 

Life with cancer is still life – let’s help people live it.",
    "themeColour": Object {
      "alpha": 255,
      "blue": 70,
      "green": 162,
      "red": 0,
    },
    "websiteUrl": "http://www.macmillan.org.uk",
  },
}
`;

exports[`fetchData actions Actions Return dataHomePage 1`] = `
Object {
  "dataHomePage": Object {
    "categories": Array [
      "Cancer",
    ],
    "countryCode": "GB",
    "currencyCode": "GBP",
    "dateAddedToJustGiving": "/Date(1036700780000+0000)/",
    "description": "At Macmillan, we know cancer can affect everything. Your body, your relationships, your money, everything. But Macmillan is here to help people live life, no matter what. We give you the support to hold on to who you are and what’s important. Life with cancer is still life –we’ll help you live it.",
    "donationDisplayAmounts": Array [
      Object {
        "amount": 3,
        "description": "Can help answer questions on cancer and give support through our helpline",
        "donationPromptType": "Monthly",
        "name": "£3.00",
      },
      Object {
        "amount": 25,
        "description": "Can fund an hour on our support line, giving vital information and support",
        "donationPromptType": "OneOff",
        "name": "£25.00",
      },
      Object {
        "amount": 5,
        "description": "Can help provide counselling to someone dealing with the impact of cancer",
        "donationPromptType": "Monthly",
        "name": "£5.00",
      },
      Object {
        "amount": 50,
        "description": "Can ensure a Macmillan Nurse is there when someone is diagnosed with cancer",
        "donationPromptType": "OneOff",
        "name": "£50.00",
      },
      Object {
        "amount": 10,
        "description": "Can help to provide the one to one medical care which matters so much",
        "donationPromptType": "Monthly",
        "name": "£10.00",
      },
      Object {
        "amount": 100,
        "description": "Can provide benefit advice so people with cancer get the best deal possible",
        "donationPromptType": "OneOff",
        "name": "£100.00",
      },
    ],
    "emailAddress": "fundraising@macmillan.org.uk",
    "id": 2116,
    "impactStatementWhat": "know cancer affects everything and exist",
    "impactStatementWhy": "help you live your life.",
    "keywords": "MCR",
    "logoAbsoluteUrl": "https://images.justgiving.com/image/e419bab6-f588-467f-a64e-5005eeb0734b.jpg",
    "logoUrl": "e419bab6-f588-467f-a64e-5005eeb0734b.jpg",
    "mobileAppeals": Array [
      Object {
        "name": "Mob Web donations 2011",
        "smsCode": "MOBL11",
      },
      Object {
        "name": "Mac In Memory Campaign",
        "smsCode": "IMEM01",
      },
      Object {
        "name": "Sussex Appeal",
        "smsCode": "SUSX13",
      },
      Object {
        "name": "Mac Social Media 2012",
        "smsCode": "MACS12",
      },
      Object {
        "name": "Translink partnership",
        "smsCode": "LINK25",
      },
      Object {
        "name": "Belfast takeover",
        "smsCode": "BELF01",
      },
      Object {
        "name": "City of London Takeover",
        "smsCode": "MACM14",
      },
      Object {
        "name": "Manchester city FR'ing",
        "smsCode": "MANC14",
      },
      Object {
        "name": "Tameside Macmillan Unit",
        "smsCode": "TAME16",
      },
    ],
    "name": "Macmillan Cancer Support",
    "pageShortName": "macmillan",
    "profilePageUrl": "https://www.justgiving.com/macmillan",
    "registrationNumber": "261017",
    "smsShortName": "Macmillan Cancer Support",
    "thankyouMessage": "Thanks so much for your donation to Macmillan Cancer Support. Your support means an incredible amount to us.

At Macmillan, we know cancer can affect everything. Your body, your relationships, your money, everything. But Macmillan is here to help people live life, no matter what. We give them the support they need to hold on to who they are and what’s important.  But we can’t do it without you. 

Life with cancer is still life – let’s help people live it.",
    "themeColour": Object {
      "alpha": 255,
      "blue": 70,
      "green": 162,
      "red": 0,
    },
    "websiteUrl": "http://www.macmillan.org.uk",
  },
}
`;
